From 4616fdae94c2f3c621d3803d58a7a5e43d78ef5a Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 30 Apr 2020 19:16:14 +0800
Subject: [PATCH] phy/rockchip: mipi-dphy: Call inno_mipi_dphy_driver_init()
 early when CONFIG_ROCKCHIP_THUNDER_BOOT=y

Before panel_simple_init().

Change-Id: I9cf38a2b6a2110b9172986d3700fa770f49e6170
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/phy/rockchip/phy-rockchip-inno-mipi-dphy.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/phy/rockchip/phy-rockchip-inno-mipi-dphy.c b/drivers/phy/rockchip/phy-rockchip-inno-mipi-dphy.c
index 77b4f8422e24..342e9eab6c70 100644
--- a/drivers/phy/rockchip/phy-rockchip-inno-mipi-dphy.c
+++ b/drivers/phy/rockchip/phy-rockchip-inno-mipi-dphy.c
@@ -831,7 +831,21 @@ static struct platform_driver inno_mipi_dphy_driver = {
 	.remove = inno_mipi_dphy_remove,
 };
 
+#ifdef CONFIG_ROCKCHIP_THUNDER_BOOT
+static int __init inno_mipi_dphy_driver_init(void)
+{
+	return platform_driver_register(&inno_mipi_dphy_driver);
+}
+fs_initcall(inno_mipi_dphy_driver_init);
+
+static void __exit inno_mipi_dphy_driver_exit(void)
+{
+	platform_driver_unregister(&inno_mipi_dphy_driver);
+}
+module_exit(inno_mipi_dphy_driver_exit);
+#else
 module_platform_driver(inno_mipi_dphy_driver);
+#endif
 
 MODULE_AUTHOR("Wyon Bi <bivvy.bi@rock-chips.com>");
 MODULE_DESCRIPTION("Innosilicon MIPI D-PHY Driver");
-- 
2.35.3

